### ISSUE TITLE
status change for "show info" toggles

### ISSUE THREAD

**Issue by nitedog** [w3c/wai-statements#65]
When "show all information" toggle is used, the information boxes are correctly expanded but the status of the individual "show info" toggles does not change accordingly.

**Comment by rvantonisse** [comment]
@nitedog This has to be fixed at [w3c/wai-website-theme](https://github.com/w3c/wai-website-theme)

@yatil you may want to implement something like a Mutation observer on single targets?

Then check whether the hidden attribute got changed/mutated inside showhidebutton target?:

```javascript
// Create an observer
observer = new MutationObserver(function (mutations, observer) {
    var hiddenMutation = mutations.filter(mutation => mutation.type === 'attributes' && mutation.attributeName === 'hidden')[0];
	console.log(`Target: ${hiddenMutation.target.id} is hidden: ${hiddenMutation.target.getAttribute('hidden') || false}`);
});

// Observe the target infobox
observer.observe(SHOWHIDEBUTTON_TARGET, { attributes: true });
```

Docs: https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver

Hope this helps! Else I am available to help on this, but I need a little help on setting up local development for w3c/wai-website-theme. (I am a beginner with Jekyll)

**Comment by yatil** [comment]
@rvantonisse Thanks, I'll look into it. I have never used a MutationObserver, but it seems to have good support. I’ll look into it either today or Wednesday. :-)

**Comment by yatil** [comment]
@nitedog @rvantonisse Can you check that it works as you expect it? Then I’ll backport the changes into the theme :-)

Also: MutationObserver is amazing and how did I not know about it?? :-)

**Comment by rvantonisse** [comment]
For Firefox, chrome, Safari I say this works great! I have not been able to test this on IE11 / Edge yet though. I think arrow functions are not supported there so that needs to be refactored to plain function. (inside the filter @ observer)

**Comment by yatil** [comment]
@rvantonisse Done.

**Comment by yatil** [comment]
@rvantonisse Also moved from an for…of loop to a normal for loop.

**Comment by rvantonisse** [comment]
@yatil Something broke it during refactor; 

I think its the missing semicolon line 91:
```javascript
return ((mutation.type === 'attributes') && (mutation.attributeName === 'hidden'))
```
Also I noticed a type Error on clicking the hide/show all button, triggered by the observer callback trying to set attribute on a button (button is null somehow, maybe an id typo?).
May be good to check if there is a button before trying to set hidden attribute on it.

**Comment by yatil** [comment]
Yeah, I have to have another look. :-) Thanks for the heads-up!

**Comment by rvantonisse** [comment]
If all is well #68 will fix the remaining errors

**Comment by yatil** [comment]
Fixed with #68! Awesome work! @nitedog See what we are doing while you are relaxing taking French public holidays? ;-)
